pkg/reconciler/resource/podspecable.go

```diff
@@ -159,3 +159,10 @@ func ConfigMapMount(name, target, cmName, cmKey string) ObjectOption {
 		})
 	}
 }
+
+func ContainerResources(resources *corev1.ResourceRequirements) ObjectOption {
+	return func(object interface{}) {
+		container := firstContainer(object)
+		container.Resources = *resources
+	}
+}
```

Comment:
The `resource` package is already quite dense. For example, we already have this:

https://github.com/triggermesh/triggermesh/blob/1e3bc1a29fc27455a55a31a8dcf2b2c9bb43aaf7/pkg/reconciler/resource/container.go#L203-L215

For things which aren't meant to be used by individual reconciler implementations but rather by a specific piece of code, I would encourage creating _local_ `ObjectOptions` instead.
In other words, move it to `pkg/reconciler/adapter.go` if you really need it, or use `resource.Requests` and `resource.Limits` instead to avoid having to maintain code that does one thing in different ways.